packages/@aws-cdk/aws-ec2/lib/security-group.ts

```diff
@@ -519,10 +525,13 @@ export class SecurityGroup extends SecurityGroupBase {
    *   strictly necessary).
    */
   private addDefaultEgressRule() {
-    if (this.allowAllOutbound) {
-      this.directEgressRules.push(ALLOW_ALL_RULE);
-    } else {
-      this.directEgressRules.push(MATCH_NO_TRAFFIC);
+    if(this.disableInlineRules) {
+      if(this.allowAllOutbound) {
+        super.addEgressRule(Peer.anyIpv4(), Port.allTraffic(), ALLOW_ALL_RULE.description, false);
+      }
```

Comment:
Unfortunately this is not good enough. We also need to add an external `MATCH_NO_TRAFFIC` rule in case of `!this.allowAlloutbound.`

Think of the following case:

* SecurityGroup is configured with `allowAllOutbound=false, disableInlineRules=true`.
* We don't add any other egress rules to the Security Group, because we want to allow NO egress at all!

CloudFormation/EC2 are going to create an SG for us and because we didn't add any outbound rules, they're going to add the default rule of "allow all traffic".

We end up with an SG that allows all traffic, but our goal was to end up with an SG that allows no traffic. Not what we wanted!

This is the reason for the default no-matching rule.

Reply:
Understood; I will add the special handling for that as discussed on the test.